libs/langchain-community/src/vectorstores/mongodb_atlas.ts

```diff
@@ -76,7 +91,18 @@ export class MongoDBAtlasVectorSearch extends VectorStore {
       [this.embeddingKey]: embedding,
       ...documents[idx].metadata,
     }));
-    await this.collection.insertMany(docs);
+    if (!this.overwrite) {
+      await this.collection.insertMany(docs);
+    } else {
+      const promises = docs.map((doc) => {
+        return this.collection.updateOne(
+          { [this.primaryKey]: doc[this.primaryKey] },
+          { $set: doc },
+          { upsert: true }
+        );
+      });
+      await Promise.all(promises);
```

Comment:
Instead of a `Promise.all` we should use the [`AsyncCaller`](https://github.com/langchain-ai/langchainjs/blob/main/langchain-core/src/utils/async_caller.ts) for some extra functionality. See[ this code example ](https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-community/src/vectorstores/typesense.ts#L134)for inspiration on how to